packages/framework-provider-azure/src/library/graphql-adapter.ts

```diff
@@ -9,16 +9,21 @@ export async function rawGraphQLRequestToEnvelope(
   const logger = getLogger(config, 'graphql-adapter#rawGraphQLRequestToEnvelope')
   logger.debug('Received GraphQL request: ', context.req)
   const requestID = context.executionContext.invocationId
-  const connectionID = undefined // TODO: Add this when sockets are supported
-  const eventType = 'MESSAGE'
+  const connectionID = context.bindingData?.connectionContext?.connectionId
+  const eventType =
+    context.bindingData?.connectionContext?.eventType !== undefined
+      ? context.bindingData?.connectionContext?.eventName?.toUpperCase()
+      : 'MESSAGE'
```

Comment:
You can use the null safe operator, `?`, when calling a function and avoid repeating the chain. Check the `?.` before the parenthesis:
```suggestion
  const eventType = context.bindingData?.connectionContext?.eventName?.toUpperCase?.() ?? 'MESSAGE'
```

Reply:
Let me refactor it a little bit to make it more readable because It's not `eventName ` OR `MESSAGE` depending on `eventName` but depending on `eventType` so the condition is ok